### ISSUE TITLE
Question: Is the JComboBox of this theme is customizable to allow usage of any component in the popup?

### ISSUE THREAD

**Issue by bric3** [JFormDesigner/FlatLaf#802]
Imagine an editable combobox, but when clicking on the drop down button the popup can contain be any component. 

Also having control on the popup anchor point might be desirable (left or right), so if the content component increase or shrink the popup will appear in place.

A bit like this ? 

<img width="191" alt="image" src="https://github.com/JFormDesigner/FlatLaf/assets/803621/6b0c252a-696a-4369-97fe-9880d94e3c7c">
<img width="300" alt="image" src="https://github.com/JFormDesigner/FlatLaf/assets/803621/9f2332c0-5002-44a2-8205-3ec448d780e0">

https://github.com/bric3/intellij-platform-swing-components

**Comment by DevCharly** [comment]
> Question: Is the JComboBox of this theme is customizable to allow usage of any component in the popup?

No, this is not supported by FlatLaf.

But seems that you've implemented something like that, right?
https://github.com/bric3/intellij-platform-swing-components/blob/main/components/src/main/kotlin/io/github/bric3/ij/components/combo/ComboBoxWithCustomPopup.kt

**Comment by bric3** [comment]
Yeah, this is an almost working attempt but not quite. I encountered bugs that seem to elude me. This code for example doesn't handle some key presses well. Also this code relies a lot on IntelliJ code, while it would be nice if a "raw" swing component supported that.

Thew biggest complaint I have with `JComboBox`, is that most of the behavior is hard coded in the `UI`, some things can be changed, but not enough.

Also this was done on my free time mostly, and I am far from a an expert on the topic, barely have 2 years of Swing experience, while I have much bigger/longer exp on some other programming topics. Which makes me second guess my approach, either in its soundness or its maintainability, yet I believe such a component might generally useful.

Like showing a calendar, possibly extending the popup with 2 months, etc.
